docs/src/main/paradox/user/lang/go/eventsourced.md

```diff
@@ -0,0 +1,93 @@
+@@@
+
+A command handler may emit an event by using the embedded `cloudstate.EventEmitter` and invoking the `Emit` method on it. Calling `Emit` will immediately invoke the associated event handler for that event - this both validates that the event can be applied to the current state, as well as updates the state so that subsequent processing in the command handler can use it.
+
+Here's an example of a command handler that emits an event:
+
+@@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/shoppingcart.go) { #add-item }
+
+This command handler also validates the command, ensuring the quantity items added is greater than zero. Returning a `error` fails the command and the support library takes care of signaling that back to the requesting proxy as a `Failure` reply.  
+
+## Handling events
+
+Event handlers are invoked at two points, when restoring entities from the journal, before any commands are handled, and each time a new event is emitted. An event handlers responsibility is to update the state of the entity according to the event. Event handlers are the only place where its safe to mutate the state of the entity at all.
+
+Event handlers are declared by either implementing the `cloudstate.EventHandler` interface
+
+@@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/event.go) { #event-handler }
+
+or implementing an unary method that matches the type of the event to be handled. Event handlers are differentiated by the type of event they handle. By default, the type of event an event handler handles will be determined by looking for a single argument that the event handler takes. If for any reason this needs to be overridden, or if the event handler method doesn't exists at all, the event is handed over to the `cloudstate.EventHandler` `Handle` method when the entity implements that interface. The by implementing the `HandleEvent(event interface{}) (handled bool, err error)` method, a event handler indicates if he handled the event or if any occurred, returns an error. The returned error has precedent and the handled flag would not be considered.  
+
+Here's an example event handler for the `ItemAdded` event.
+
+@@snip [shoppingcart.go](/docs/src/main/paradox/user/lang/go/src/shoppingcart.go) { #item-added }
+
+## Producing and handling snapshots
+
+## Multiple behaviors
+
+Multiple behaviors are not supported yet by the Go support library. 
+
+## Registering the entity
+
+Once you've created your entity, you can register it with the `cloudstate.Cloudstate` server, by invoking the `Register` method of an Cloudstate instance. In addition to passing your entity type and service name, you also need to pass any descriptors that you use for persisting events, for example, the `domain.proto` descriptor.
+
+During registration the oprtional ServiceName and the ServiceVersion can be configured as Options.
```

Comment:
```suggestion
During registration the optional ServiceName and the ServiceVersion can be configured as Options.
```

Perhaps also give an example on how to pick values for these?

Reply:
- [ ] fix typo
- [ ] give an example on how to pick values for these